src/vm/hillclimbing.cpp

```diff
@@ -335,7 +335,7 @@ int HillClimbing::Update(int currentThreadCount, double sampleDuration, int numC
     // we'll simply stay at minThreads much longer, and only occasionally try a higher value.
     //
     if (ratio.r < 0.0 && newThreadCount == ThreadpoolMgr::MinLimitTotalWorkerThreads)
-        *pNewSampleInterval = (int)(0.5 + m_currentSampleInterval * (10.0 * max(-ratio.r, 1.0)));
+        *pNewSampleInterval = (int)(0.5 + m_currentSampleInterval * (10.0 * min(-ratio.r, 1.0)));
```

Comment:
In addition to the overflow concerns, what kind of impact was the max vs min having on actual thread growth?  I'm wondering if this might actually have substantial implications and need some broader testing on a variety of scenarios.

Reply:
The thread growth calculation was doing the proper min/max above this code to limit ratio.r to {-1..1}. An effect of this issue is that the new sample interval becomes int32_min after overflow (exception on overflow is disabled by default) and it's added to a GetTickCount() value to determine the next threshold for sampling (stored as unsigned), so the interval becomes very large and hill climbing gets disabled.

Reply:
> The thread growth calculation was doing the proper min/max above this code to limit ratio.r to {-1..1}.

Ah. Ok.